src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -70,11 +72,18 @@ internal sealed class MemberExpansion : Expansion
             var includeInherited = (flags & ExpansionFlags.IncludeBaseMembers) == ExpansionFlags.IncludeBaseMembers;
             var hideNonPublic = (inspectionContext.EvaluationFlags & DkmEvaluationFlags.HideNonPublicMembers) == DkmEvaluationFlags.HideNonPublicMembers;
             var includeCompilerGenerated = (inspectionContext.EvaluationFlags & DkmEvaluationFlags.ShowValueRaw) == DkmEvaluationFlags.ShowValueRaw;
-            runtimeType.AppendTypeMembers(allMembers, predicate, declaredTypeAndInfo.Type, appDomain, includeInherited, hideNonPublic, isProxyType, includeCompilerGenerated);
+            var favoritesInfo = (supportsFavorites) ? type.GetFavorites() : null;
+            runtimeType.AppendTypeMembers(allMembers, predicate, declaredTypeAndInfo.Type, appDomain, includeInherited, hideNonPublic, isProxyType, includeCompilerGenerated, supportsFavorites, favoritesInfo);
+
+            var favoritesMembersByName = new Dictionary<string, MemberAndDeclarationInfo>();
 
             foreach (var member in allMembers)
             {
-                if (member.IsStatic)
+                if (member.IsFavorite)
+                {
+                    favoritesMembersByName.Add(member.Name, member);
+                }
+                else if (member.IsStatic)
```

Comment:
>else if [](start = 16, length = 7)

Does this mean we do not duplicate favorites when listing regular members? Correct?

Reply:
Yes, favorites "move" to the top of the list.